src/js/components/TextArea/TextArea.js

```diff
@@ -8,6 +8,7 @@ import { StyledTextArea } from './StyledTextArea';
 
 class TextArea extends Component {
   onEsc = event => {
+    event.stopPropagation();
```

Comment:
Why was this needed?

Reply:
take a look here
https://storybook.grommet.io/?selectedKind=Layer&selectedStory=Form&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel

go to the input and try pressing esc inside the layer. the layer will close.

I don't think we should be using stopImmediatePropagation.

Reply:
The Layer isn't closing for me in the Layer Form story.